app/javascript/components/UserAssignmentForm.js

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import UserAssignmentFormField from "./UserAssignmentFormField";
+
+class UserAssignmentForm extends React.Component {
+  constructor(props) {
+    super();
+    this.state = props;
+    this.handleAddClick = this.handleAddClick.bind(this);
+    this.handleDeleteClick = this.handleDeleteClick.bind(this);
+  }
+
+  randomUUID() {
+    return "new-mission-"+ Math.floor(Math.random() * Math.floor(9000));
+  }
+
+  handleAddClick() {
+    let assignments = this.state.assignments;
+    assignments.push({role: "", mission: this.randomUUID(), new_assignment: true});
```

Comment:
A comment here on what the `randomUUID` for would be nice since this is not the actual ID of any mission or whatever. Kind of confusing. I think you're using it as a key for React only? But not sure.